src/providers/v2/pool-provider.ts

```diff
@@ -144,7 +164,39 @@ export class V2PoolProvider implements IV2PoolProvider {
         continue;
       }
 
-      const [token0, token1] = sortedTokenPairs[i]!;
+      let [token0, token1] = sortedTokenPairs[i]!;
+      if (
```

Comment:
lets make this a function because these two snippets are exactly the same :)

Reply:
agree, will follow-up separately.